### DISCUSSION TITLE
Plotting TRAINING DICE and VALIDATION LOSS in tensorboard

### DISCUSSION THREAD

**Discussion by vlgonz** [Project-MONAI/tutorials#1659] [General]
Hi, I am using monai event handlers, specifically tensorboard, to plot training loss and validation dice score. Besides, I have seen a couple of examples in monai notebooks provided in this github. However, I haven't found any information about how to plot training dice and validation loss using tensorboard handler, can anyone provide an example on how to do this? Thank you very much.

**Comment by KumoLiu** [comment]
Hi @vlgonz, by default, 'TensorBoardStatsHandler' plots loss and metric; therefore, you must ensure that the metric and loss are calculated in the correct engine.
https://github.com/Project-MONAI/MONAI/blob/e9e273858b6904659b0cd581d357662dc5c782f6/monai/handlers/tensorboard_handlers.py#L79-L83
You can set `output_transform=lambda x: x` to plot loss.
https://github.com/Project-MONAI/tutorials/blob/570b19e678e06aba26f53df8bc73f848ae2984ba/3d_classification/ignite/densenet_training_dict.py#L149

Hope it help, thanks.

**Reply by vlgonz** [reply]
Okay, thank you for the answer. I have already tried that (commented lines below) but it does not work. This is the code part related to validation and train ignite handlers, any different idea on how to log validation loss and training dice? 

val_handlers = [
        # TensorBoardStatsHandler(log_dir="../runs/Mar11_03-16_00/val/graphs/FOLD_{}".format(index),
        #     tag_name="val_loss", output_transform=lambda x:x),

        TensorBoardStatsHandler(log_dir="../runs/Mar11_03-16_00/val/graphs/FOLD_{}".format(index),
            output_transform=lambda x:None, global_epoch_transform=lambda x: evaluator.state.epoch),
        CheckpointSaver(save_dir="../runs/Mar11_03-16_00/val/model_checkpoints/FOLD_{}".format(index), save_dict={"net": net}, save_key_metric=True)]

    evaluator = SupervisedEvaluator(
        device=device,
        val_data_loader=val_loaders[index],
        network=net,
        inferer=SlidingWindowInferer(roi_size=(64,64,64), sw_batch_size=4, overlap=0.5, progress=True),
        postprocessing=val_post_transforms,
        key_val_metric={
            "val_mean_dice": MeanDice(include_background=False,output_transform=from_engine(["pred", "label"]))},
        val_handlers=val_handlers)

    train_handlers = [

        ValidationHandler(validator=evaluator, interval=1, epoch_level=True),
        TensorBoardStatsHandler(log_dir="../runs/Mar11_03-16_00/train/graphs/FOLD_{}".format(index), tag_name="train_loss", output_transform=from_engine(["loss"], first=True))]

    trainer = SupervisedTrainer(
        device=device,
        max_epochs=100,
        train_data_loader=train_loaders[index],
        network=net,
        optimizer=opt,
        loss_function=loss,
        inferer=SimpleInferer(),
        train_handlers=train_handlers,
    )
